pennylane/interfaces/batch/jax.py

```diff
@@ -1,299 +1,332 @@
-            new_tapes = []
-
-            for t, a in zip(tapes, p):
-                new_tapes.append(t.copy(copy_operations=True))
-                new_tapes[-1].set_parameters(a)
-
-            res, jacs = execute_fn(new_tapes, **gradient_kwargs)
-
-            # On the forward execution return the jacobian too
-            return res, jacs
-
-        fwd_shapes = [jax.ShapeDtypeStruct((1,), dtype) for _ in range(total_size)]
-        jacobian_shape = [jax.ShapeDtypeStruct((1, len(p)), dtype) for p in params]
-        res, jacs = host_callback.call(
-            wrapper,
-            params,
-            result_shape=tuple([fwd_shapes, jacobian_shape]),
-        )
-        return res, jacs
-
-    def wrapped_exec_fwd(params):
-        res, jacs = wrapped_exec(params)
-        return res, tuple([jacs, params])
-
-    def wrapped_exec_bwd(params, g):
-
-        # Use the jacobian that was computed on the forward pass
-        jacs, params = params
-
-        # Adjust the structure of how the jacobian is returned to match the
-        # non-forward mode cases
-        # E.g.,
-        # [DeviceArray([[ 0.06695931,  0.01383095, -0.46500877]], dtype=float32)]
-        # is mapped to
-        # [[DeviceArray(0.06695931, dtype=float32), DeviceArray(0.01383095,
-        # dtype=float32), DeviceArray(-0.46500877, dtype=float32)]]
-        res_jacs = []
-        for j in jacs:
-            this_j = []
-            for i in range(j.shape[1]):
-                this_j.append(j[0, i])
-            res_jacs.append(this_j)
-        return tuple([tuple(res_jacs)])
-
-    wrapped_exec.defvjp(wrapped_exec_fwd, wrapped_exec_bwd)
-    res = wrapped_exec(params)
-
-    tracing = any(isinstance(r, jax.interpreters.ad.JVPTracer) for r in res)
-
-    # When there are no tracers (not differentiating), we have the result of
-    # the forward pass and the jacobian, but only need the result of the
-    # forward pass
-    if len(res) == 2 and not tracing:
-        res = res[0]
-
-    return res
+# Copyright 2018-2022 Xanadu Quantum Technologies Inc.
+
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+
+#     http://www.apache.org/licenses/LICENSE-2.0
+
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""
+This module contains functions for adding the JAX interface
+to a PennyLane Device class.
+"""
+# pylint: disable=too-many-arguments
+import jax
+import jax.numpy as jnp
+
+import pennylane as qml
+from pennylane.operation import Sample, Probability
+
+dtype = jnp.float64
+
+
+def _get_jax_interface_name(tapes):
```

Comment:
@antalszava since this is imported and used in another file, I suggest making it public